cpmpy/tools/tune_solver.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+
+from ..solvers.utils import SolverLookup, param_combinations
+from ..solvers.solver_interface import ExitStatus
+
+class ParameterTuner:
+
+    def __init__(self, solvername, model, all_params=None, defaults=None):
+        """
+            Parameter tuner based on DeCaprio method [ref_to_decaprio]
```

Comment:
you really have to insert publication details before we merge.

Ideally at top of file, something like
"""
   Parameter tuning with adaptive capping and...

   Based on the paper "..." by ... at ...
   - DOI: url with DOI (ppl can find the bibtex there)
   - paper: link to pdf
   - code used in the paper: link

   The code currently implements ...